### PR TITLE
PT-1389 Generate soup doc

### PR DESCRIPTION
**Pull request by franck-fourel**
Add script to generate basic soup documentation
Dependency on https://github.com/mdblp/tools/pull/19

__Note__: the branch is pointing to the PR branch defined in https://github.com/mdblp/tools/pull/19

### CODE REVIEW

File: artifact.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#!/bin/sh -e
+#!/bin/sh -ex
 
-wget -q -O artifact_go.sh 'https://raw.githubusercontent.com/mdblp/tools/dblp/artifact/artifact_go.sh'
+wget -q -O artifact_go.sh 'https://raw.githubusercontent.com/mdblp/tools/feature/add-golang-soup/artifact/artifact_go.sh'
```
Comment (BigFab) [inline artifact.sh:3]: this one can be removed now

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,4 @@ vendor/
 
 *.exe
 artifact_go.sh
+soup/
```
Comment (BigFab) [inline .gitignore]: Can we push to "docs" folder like for the openapi documentation?

File: buildSoup.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh -eu
+# Generate Soups
+
+DEPLOY_DOC=${DEPLOY_DOC:-soup}
+TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG:-owner/repo}
+APP="${TRAVIS_REPO_SLUG#*/}"
+GO111MODULE=on
+
+if [ -z "${TRAVIS_TAG+x}" ]; then
```
Comment (BigFab) [inline buildSoup.sh:9]: typo?

File: buildSoup.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  TRAVIS_TAG=0.0.0
+fi
+if [ -d ${DEPLOY_DOC} ]; then
+  rm -rf ./${DEPLOY_DOC}
```
Comment (BigFab) [inline buildSoup.sh:13]: if we push to "docs" this deletion might have impacts on the generation of the openapi documentation in the other step

**Review by BigFab — CHANGES_REQUESTED** [review]
I think we can merge like that after the modification 1. for using the master branch of the `tools` repository and 2. the typo '+x' (?)
I will make another PR to generate the openapi documentation for Shoreline so we will have both SOUPs and OPENAPI documentation generation. This might need some adjustments I can do in the coming PR

File: artifact.sh (hunk 1 above)
Comment (franck-fourel) [reply artifact.sh:3]: done

File: buildSoup.sh (hunk 3 above)
Comment (franck-fourel) [reply buildSoup.sh:9]: you mean the +x ?
it's a way of testing that TRAVIS_TAG is not set

File: .gitignore (hunk 2 above)
Comment (franck-fourel) [reply .gitignore]: sounds ok

File: buildSoup.sh (hunk 3 above)
Comment (BigFab) [reply buildSoup.sh:9]: ok then

**Review by BigFab — APPROVED** [review]
Very good :)

**Pull request merged**